apps/website-25/src/components/courses/exercises/Exercise.tsx

```diff
@@ -12,23 +15,64 @@ type ExerciseProps = {
 const Exercise: React.FC<ExerciseProps> = ({
   exerciseId,
 }) => {
-  const [{ data }] = useAxios<GetExerciseResponse>({
+  const exerciseClassNames = 'exercise not-prose my-6';
+
+  const auth = useAuthStore((s) => s.auth);
+
+  const [{ data: exerciseData }] = useAxios<GetExercise>({
     method: 'get',
     url: `/api/courses/exercises/${exerciseId}`,
   });
 
-  if (!data || !data.exercise) {
+  const [{ data: responseData }, refetch] = useAxios<GetExerciseResponseResponse>({
+    method: 'get',
+    url: `/api/courses/exercises/${exerciseId}/response`,
+    headers: auth ? {
+      Authorization: `Bearer ${auth.token}`,
+    } : undefined,
+  });
+
+  const handleExerciseSubmit = useCallback(async (exerciseResponse: string, completed?: boolean) => {
+    await axios.put<unknown, unknown, PutExerciseResponseRequest>(
+      `/api/courses/exercises/${exerciseId}/response`,
+      {
+        response: exerciseResponse,
+        completed,
+      },
+      {
+        headers: {
+          Authorization: `Bearer ${auth?.token}`,
+        },
+      },
+    );
+
+    await refetch();
```

Comment:
Does this definitely work well if the user is on a slow connection? I'm kind of worried about:
- user writes some text
- hits save, and the previous response starts refetching...
- writes some additional text
- their slow internet loads their response and deletes their additional text

I don't know if the above actually does happen on the refetch.

An alternative to avoid this might be to have a state variable which we initially set to the fetched value (on first component load), but then update when they want to save (as the first action, before sending it to the server). It saves us a network call and means the above problem doesn't occur.